java-frontend/src/main/java/org/sonar/java/se/checks/OptionalGetBeforeIsPresentCheck.java

```diff
@@ -130,22 +149,33 @@ public IsPresentSymbolicValue(SymbolicValue sv) {
      * Will be called only after calling Optional.isPresent()
      */
     @Override
-    public List<ProgramState> setConstraint(ProgramState programState, BooleanConstraint booleanConstraint) {
-      OptionalConstraint optionalConstraint =  programState.getConstraint(optionalSV, OptionalConstraint.class);
+    public List<ProgramState> setConstraint(ProgramState ps, BooleanConstraint booleanConstraint) {
+      ProgramState programState = ps;
+      OptionalConstraint optionalConstraint = programState.getConstraint(optionalSV, OptionalConstraint.class);
       if (isImpossibleState(booleanConstraint, optionalConstraint)) {
         return Collections.emptyList();
       }
       if (optionalConstraint == OptionalConstraint.NOT_PRESENT || optionalConstraint == OptionalConstraint.PRESENT) {
         return Collections.singletonList(programState);
       }
       OptionalConstraint newConstraint = booleanConstraint.isTrue() ? OptionalConstraint.PRESENT : OptionalConstraint.NOT_PRESENT;
+
+      if (newConstraint == OptionalConstraint.PRESENT && optionalSV instanceof FilteredOptionalSymbolicValue) {
+        programState = ((FilteredOptionalSymbolicValue) optionalSV).propagatePresentConstraint(programState);
```

Comment:
I'm not sure if there is a better way to do this, but I couldn't find any existing method allowing to do something similar to this `propagatePresentConstraint`

Reply:
Why not override the setConstraint method in OptionalSV for this purpose ?